### PR TITLE
Cloud network path APIs

### PR DESCRIPTION
**Pull request by dstiliadis**
Adds the cloud network path data structures.

### CODE REVIEW

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cloudpath",
    "component": "gaia",
    "pr-id": "695",
    "commit-sha": "9416639d04555a1511ca74609bd45e433f74df0b",
    "pipeline": "master"
  },
  {
    "project": "cloudpath",
    "component": "backend",
    "pr-id": "1312",
    "commit-sha": "5968a81352e1c9cf2c13eaf47ed6b1b31b2013fd",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cloudpath",
    "component": "gaia",
    "pr-id": "695",
    "commit-sha": "a62e13393350ba05e210d7bba29b0f6fd40d799b",
    "pipeline": "master"
  },
  {
    "project": "cloudpath",
    "component": "backend",
    "pr-id": "1312",
    "commit-sha": "13da532364e7d7064ebf4c23c69483ba935aba35",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cloudpath",
    "component": "gaia",
    "pr-id": "695",
    "commit-sha": "a62e13393350ba05e210d7bba29b0f6fd40d799b",
    "pipeline": "master"
  },
  {
    "project": "cloudpath",
    "component": "backend",
    "pr-id": "1312",
    "commit-sha": "6b5fae0317a64d89cdda725e1d3e8edddd689d15",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cloudpath",
    "component": "gaia",
    "pr-id": "695",
    "commit-sha": "a62e13393350ba05e210d7bba29b0f6fd40d799b",
    "pipeline": "master"
  },
  {
    "project": "cloudpath",
    "component": "backend",
    "pr-id": "1312",
    "commit-sha": "944ceebee0953b0a09532d46535b74dde47ee5ef",
    "pipeline": "master"
  }
]
```

File: custom_validations.go [hunk 1]
```diff
@@ -1547,6 +1547,24 @@ func ValidateCloudNetworkQueryEntity(q *CloudNetworkQuery) error {
...
+		if len(q.DestinationIP) != 0 && len(q.SourceSelector.ObjectIDs) != 1 {
+			return makeValidationError("Entity CloudNetworkQuery", "Only one entity (interface/instance) must be selected for the source selector for network path queries")
+		}
+
+		if len(q.SourceIP) == 0 && len(q.SourceSelector.ObjectIDs) != 1 {
+			return makeValidationError("Entity CloudNetworkQuery", "Only one entity (interface/instance) must be selected for the source selector for network path queries")
+		}
+
+		if len(q.DestinationIP) == 0 && len(q.DestinationSelector.ObjectIDs) != 1 {
+			return makeValidationError("Entity CloudNetworkQuery", "Only one entity (interface/instance) must be selected for the source selector for network path queries")
```
Comment (jping0929) [inline custom_validations.go:1564]: Typo? Should it be `Only one entity (interface/instance) must be selected for the destination selector`

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "cloudpath",
    "component": "gaia",
    "pr-id": "695",
    "commit-sha": "254c9cad71db1a0b2e031581ab13caed1ce23873",
    "pipeline": "master"
  },
  {
    "project": "cloudpath",
    "component": "backend",
    "pr-id": "1312",
    "commit-sha": "b18990a1c6cfe8c1f791c8e69d61b6912536d528",
    "pipeline": "master"
  }
]
```

**Pull request merged**